pkg/crc/machine/vsock.go

```diff
@@ -146,3 +139,39 @@ func getSSHTunnelURI() string {
 	}
 	return u.String()
 }
+
+func HTTPAndHTTPS() []types.ExposeRequest {
+	hPort, hsPort := httpPort, httpsPort
+	if isPortInUse(hPort) {
+		logging.Warnf("Port %s is already in use. Going to use %s", httpPort, alternateHTTPPort)
+		hPort = alternateHTTPPort
+	}
+
+	if isPortInUse(hsPort) {
+		logging.Warnf("Port %s is already in use. Going to use %s", httpsPort, alternateHTTPSPort)
+		hsPort = alternateHTTPSPort
+	}
+
+	return []types.ExposeRequest{
+		{
+			Protocol: "tcp",
+			Local:    fmt.Sprintf(":%s", hsPort),
+			Remote:   net.JoinHostPort(virtualMachineIP, httpsPort),
+		},
+		{
+			Protocol: "tcp",
+			Local:    fmt.Sprintf(":%s", hPort),
+			Remote:   net.JoinHostPort(virtualMachineIP, httpPort),
+		},
```

Comment:
I'd have a preference for *not* putting http/https together to avoid code duplication.
```
exposeRequest = append(exposeRequest, requestWithAlternatePort(httpPort, alternateHTTPPort)
exposeRequest = append(exposeRequest, requestWithAlternatePort(httpsPort, alternateHTTPSPort)
```

Reply:
`Alternate` is not the right wording as that means that either can be assigned consecutively.

Reply:
`fallback` would work.